server-common/src/main/java/org/apache/kafka/server/util/timer/TimerTaskEntry.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.util.timer;
+
+public class TimerTaskEntry {
```

Comment:
Missing compareTo method.

A unit test should have caught this. Please add a unit test if we are having a gap in testing.

Reply:
You meant `compare`, right? We can't implement in Java because it comes from Scala trait. It was only used by `MockTimer` and I have refactored it to not require it any more.

Reply:
By extending `Ordered` in scala for TimerTaskEntry, we wanted to specify that objects of this class are comparable (i.e. functions such as sprt can be used) and should be compared based on their expirationMs. We should implement a similar mechanism in Java using Comparable over here https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html

Reply:
As explained in the other comment, this is not required any more.